pulsar/producer_partition.go

```diff
@@ -298,7 +305,9 @@ func (p *partitionProducer) reconnectToBroker() {
 		d := backoff.Next()
 		p.log.Info("Reconnecting to broker in ", d)
 		time.Sleep(d)
-
+		atomic.AddUint64(&p.epoch, 1)
+		p.log.WithField("epoch", atomic.LoadUint64(&p.epoch)).Debug(
```

Comment:
This will log the epoch twice once as a field and once as a message. It might make more sense to add the epoch log after the producer is created? Curious what's the significance using and knowing the epoch?

Reply:
Updated, the reason to log the epoch is mainly for debug.